tangos/input_handlers/yt.py

```diff
@@ -85,6 +170,130 @@ def _load_halo_cat_without_caching(self, ts_extension, snapshot_file):
+            filter = data[("all", "particle_family")] == 2 # DM = 1, Stars = 2
+            return filter
+
+        add_particle_filter("stars", function=Stars, filtered_type='all', \
+                            requires=["particle_family"])
+
+        def AllDarkMatter(pfilter, data):
+            #filter = np.logical_or(data[("all", "particle_type")] == 4,data[("all", "particle_type")] == 1) # DM = 1, Stars = 2
+            filter = data[("all", "particle_family")] == 1
+            return filter
+
+        add_particle_filter("dark_matter", function=AllDarkMatter, filtered_type='all', \
+                            requires=["particle_family"])
+
+        def MustRefineParticles(pfilter, data):
+            filter = data[("all", "particle_family")] == 4
+            return filter
+
+        add_particle_filter("mrp_dark_matter", function=MustRefineParticles, filtered_type='all', \
+                            requires=["particle_family"])
+
+        f.add_particle_filter("stars")
+        f.add_particle_filter("dark_matter")
+        f.add_particle_filter("mrp_dark_matter")
+
+        return f
+
+    def _load_halo_cat_without_caching(self, ts_extension, snapshot_file):
+        # Check whether datasets.txt exists (i.e., if rockstar was run with yt)
+        if os.path.exists(self._extension_to_filename("datasets.txt")):
+            fnum = read_datasets(self._extension_to_filename(""),ts_extension)
+        else: # otherwise, assume a one-to-one correspondence
+            overdir = self._extension_to_filename("")
+            snapfiles = glob.glob(overdir+ts_extension[:2]+len(ts_extension[2:].split('/')[0])*'?')
+            rockfiles = glob.glob(overdir+"out_*.list")
+            sortind = np.array([int(rname.split('.')[0].split('_')[-1]) for rname in rockfiles])
+            sortord = np.argsort(sortind)
+            snapfiles.sort()
+            rockfiles = np.array(rockfiles)[sortord]
+            timestep_ind = np.argwhere(np.array([s.split('/')[-1] for s in snapfiles])==ts_extension.split('/')[0])[0]
+            fnum = int(rockfiles[timestep_ind][0].split('.')[0].split('_')[-1])
+        cat = yt.load(self._extension_to_filename("halos_"+str(fnum)+".0.bin"))
+        cat_data = cat.all_data()
+        # Check whether rockstar was run with Behroozi's distribution or Wise's
+        if np.any(cat_data["halos","particle_identifier"]<0):
+            del cat
+            del cat_data
+            cat = yt.load(self._extension_to_filename("halos_"+str(fnum)+".0.bin"))
+            cat.parameters['format_revision'] = 2 #
+            cat_data = cat.all_data()
+        return cat, cat_data
+
+    def enumerate_objects(self, ts_extension, object_typetag="halo", min_halo_particles=config.min_halo_particles):
+        if object_typetag!="halo":
+            return
+        if self._can_enumerate_objects_from_statfile(ts_extension, object_typetag):
+            yield from self._enumerate_objects_from_statfile(ts_extension, object_typetag)
```

Comment:
I'd suggest returning after the yield, something like:
```
            yield from self._enumerate_objects_from_statfile(ts_extension, object_typetag)
            return
```
This way you can de-indent the rest of the function which is always nice :)